### PR TITLE
Set default external name config as IdentifierFromProvider and a common referencer for project_id

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

This PR:

- Configures default external name config as `IdentifierFromProvider`.
- Configures a common reference for `project_id` field that applies to all resources.
- Some refactoring for resource defaulting

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I didn't test but only run `make generate`.

I quickly skimmed through the resources in the [Terraform provider](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs) and it looks like `IdentifierFromProvider` would better fit and bring the resources in a working state.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: config/mongodbatlas/config.go
Comment (ulucinar) [inline config/mongodbatlas/config.go:56]: I think we will need to specify the package here.

Comment (turkenh) [reply config/mongodbatlas/config.go]: Thanks @ulucinar, that was the problem failing the CI.

**Comment by bitgandtter** [comment]
Can we please add the same config for users? I'm having the same issue. This message is just to replicate what I mention over slack. Also I'm pretty sure that will be a recurrent issue with other resources

**Comment by turkenh** [comment]
> Can we please add the same config for users? I'm having the same issue. This message is just to replicate what I mention over slack. Also, I'm pretty sure that will be a recurrent issue with other resources

Just pushed a change that configures all resources for, which would include user resource
- External name as identifierFromProvider
- Adds referencer for `project_id` field

**Review by ulucinar — APPROVED** [review]
I have left some nitpicks but I think we okay with this. Once we merge this I can release a new version. Thank you very much.

**Comment by bitgandtter** [comment]
Any eta i really need this one please

**Comment by bitgandtter** [comment]
@ulucinar can this be merged please im blocked and I really want to use this provider. thanks so much in advance

**Comment by turkenh** [comment]
> @ulucinar can this be merged please im blocked and I really want to use this provider. thanks so much in advance

@bitgandtter just noticed that there are some comments that need to be resolved, sorry for that. 
I'll work on them and try to get this merged ASAP.

File: config/overrides.go
Comment (turkenh) [reply config/overrides.go:13]: Ah good call, thanks!

**Comment by abdelazizhd** [comment]
Hi, we are waiting for this merge, any expected date or is there any other problem? Thanks

**Comment by ulucinar** [comment]
Thanks @turkenh, I will cut the `v0.2.0` release tonight!

**Comment by ulucinar** [comment]
@abdelazizhd & @bitgandtter, you can give `crossplane/provider-jet-mongodbatlas:v0.2.0` consuming this PR a try! Thank you all!

**Comment by bitgandtter** [comment]
@ulucinar got this after trying
```
observe failed: cannot run refresh: readObjectStart: expect { or n, but
        found S, error found in #1 byte of ...|Stack trace|..., bigger context
        ...|Stack trace from the terraform-provider-mongodbatla|...: refresh
        failed
``

**Pull request merged**
